glidex.forms/GlideExtensions.cs

```diff
@@ -86,6 +89,100 @@ public static async Task LoadViaGlide (this ImageView imageView, ImageSource sou
 			}
 		}
 
+		public static async Task<Bitmap> LoadViaGlide (this ImageSource source, Context context, CancellationToken token)
+		{
+			try {
+				if (!IsActivityAlive (context, source))
+					return null;
+
+				RequestManager request = With (context);
+				RequestBuilder builder = null;
+
+				if (source is null) {
+					Forms.Debug ("`{0}` is null", nameof (ImageSource));
+					return null;
+				}
+
+				switch (source) {
+					case FileImageSource fileSource:
+						var fileName = fileSource.File;
+						var drawable = ResourceManager.GetDrawableByName (fileName);
+						if (drawable != 0) {
+							Forms.Debug ("Loading `{0}` as an Android resource", fileName);
+							builder = request.AsBitmap ().Load (drawable);
+						} else {
+							Forms.Debug ("Loading `{0}` from disk", fileName);
+							builder = request.AsBitmap ().Load (fileName);
+						}
+						break;
+
+					case UriImageSource uriSource:
+						var url = uriSource.Uri.OriginalString;
+						Forms.Debug ("Loading `{0}` as a web URL", url);
+						builder = request.AsBitmap ().Load (url);
+						break;
+
+					case StreamImageSource streamSource:
+						Forms.Debug ("Loading `{0}` as a byte[]. Consider using `AndroidResource` instead, as it would be more performant", nameof (StreamImageSource));
+						using (var memoryStream = new MemoryStream ())
+						using (var stream = await streamSource.Stream (token)) {
+							if (token.IsCancellationRequested || stream == null)
+								return null;
+							if (!IsActivityAlive (context, source)) {
+								return null;
+							}
+							await stream.CopyToAsync (memoryStream, token);
+							builder = request.AsBitmap ().Load (memoryStream.ToArray ());
+						}
+						break;
+				}
```

Comment:
This long `switch` statement is a lot of copy-paste. This should be refactored out into smaller methods that the other `LoadViaGlide()` method can share.